docs/guides/templating.md

```diff
@@ -2,48 +2,54 @@
 
 With External Secrets Operator you can transform the data from the external secret provider before it is stored as `Kind=Secret`. You can do this with the `Spec.Target.Template`. Each data value is interpreted as a [golang template](https://golang.org/pkg/text/template/).
 
+!!! note
+
+    **.'spec.data.secretkey'** cannot contain **'-'** (dashes), consider using camelcase when defining the secret keys.
```

Comment:
Good suggestion, this definitely will help people! Though technically you can use dashes, it's just a pain in the ass to work with :sweat_smile: 

Something along the lines of:
```
{{  index . "service-account-token" }}
```

Reply:
Adjusted a bit to add a note around using the index function